### DISCUSSION TITLE
locking activity after a parallel foreach

### DISCUSSION THREAD

**Discussion by Maroua90** [elsa-workflows/elsa-core#2965] [Q&A]
hello, I have a blocking activity after a parallel foreach which is waiting for an input to be restarted, on the other hand the column [WorkflowInstances].[Data] there is only the last activity blocked and not all the activities blocked after the Paralleforeach also DispatchWorkflowsAsync return only the last activity. any idea ?

**Comment by sfmskywalker** [comment]
Yes, the `ParallelForEach` activity doesn't behave as expected when using blocking activities, unfortunately.
The same issue has been reported [here](https://github.com/elsa-workflows/elsa-core/issues/1025) and a fix has been proposed [here](https://github.com/elsa-workflows/elsa-core/issues/1059).

Elsa 3 (currently under development)  has scoped contexts natively, so it shouldn't be a problem there, but we'll have to test for it to be sure.
